kartothek/io/dask/_update.py

```diff
@@ -35,6 +37,40 @@ def _hash_bucket(df: pd.DataFrame, subset: List[str], num_buckets: int):
     return df.assign(**{_KTK_HASH_BUCKET: buckets.astype(f"uint{bit_width}")})
 
 
+def _pack_payload(partition: pd.DataFrame, group_key: List[str]) -> pd.DataFrame:
```

Comment:
I think we could use some short explanation either in the `_[un]pack_payload` docstrings or on the module-level (top-docstring) that explains what's going on there. Esp. replacing the entire group DF with a single row and packing the content into a binary columns only makes sense to me after reading it twice. I could also volunteer to do some ascii-art-drawing / example stuff if you want. I just think the entire module actually uses a quite simple / streamlined approach (which is nice) but from just reading the code, it requires substantial brain power to understand it.

Reply:
I would just post an example like

pd.DataFrame({"group_col": [1, 2], "payloadA": ..., "payloadB": ...})

becomes

pd.DataFrame({"group_col": [1, 2], "payload_bytes": b"long byte string"})


if your internal ascii artist comes up with something nicer, I wouldn't mind of course :)

Reply:
I can also add a module doc string and also explain the bucketing/hashing to wrap things up

Reply:
I think an example is already very very good, because it's something most people can relate to :)

Reply:
Added documentation and type annotations to the most relevant functions. Also added an explicit example how the packing will impact a partition.